### ISSUE TITLE
GRU implementation

### ISSUE THREAD

**Issue by rongzha1** [samsungsds-rnd/deepspeech.mxnet#13]
Hi,

I want to do some optimization for this model using mkldnn backend library.

But I find that GRU.py  seems not implemented as standardized one. The difference is that Batch Norm for i2h is used in this model after indata*weight+bias.

Can you kindly show the reference paper ?

Thanks very much.

**Comment by Soonhwan-Kwon** [comment]
I referenced deep speech 2 paper and following link may help, the author of deep speech 2 wrote a comment on batch normalization. https://www.reddit.com/r/MachineLearning/comments/4082yn/rnn_batch_normalization_for_grulstm/?st=JBEAKSUQ&sh=14046726

**Comment by Soonhwan-Kwon** [comment]
And also we referenced BatchNormalized Recurrent Neural Net also. Maybe it is what you're looking for.
https://arxiv.org/pdf/1510.01378.pdf

**Comment by rongzha1** [comment]
Thanks.  We are going to support standard GRU in mkldnn first, then batchNorm GRU.
